_includes/salary-tracker.html

```diff
@@ -161,16 +180,55 @@ <h5 class="mt-5 salary-tracker-info-label">Salary Brackets</h5>
     function updatePosition(position) {
 
         selectedPosition = position;
-        updateData(null, position);
+        updateData(null, position, null, 1);
+
+    }
+
+    // Update the selected department
+    function updateDepartment(department) {
+
+        selectedDepartment = department;
+        updateData(null, null, department, 0);
 
     }
 
+    // Populate select options for Roles and Departments
+    // This also acts as a filter for Roles based on selected department
+    // Update option can have 4 values - 0: Department was updated; 1: Position was updated; 2: Reset positions; null: don't do anything 
+    function updateSelectOptions(updateOption) {
+        var countryArr = dataArr.filter(obj => {
+            return obj.country === selectedCountry;
+        });
+        if (updateOption === 0) {
+            removeSelectOptions(selectRoleEle);
+            selectedPosition = null;
+        }
+
+        if (countryArr[0].positions[0] !== undefined) {
+            countryArr[0].positions.forEach(position => {
```

Comment:
Don't forget to check if countryArr and positions is greater than 0 :) possible array out of bounds ?